docs/DeveloperGuide.md

```diff
@@ -241,16 +241,90 @@ Below is a class diagram describing the implementation of `Task` and its respect
-- Pros: Provides flexibility in how users can create a `GradedTest` object.
-- Cons: Requires additional parsing. Different constructor will give different error messages.
+- Pros:
+  * Flexibility: This choice offers flexibility in representing the graded tests, allowing for customization (e.g. usage of default value, `-`).
+  * Clarity: It will be 
+- Cons:
+  * Parsing Overhead: It requires additional parsing to generate statistics of how the Avenger's students are doing (e.g average,
+    highest, lowest score).
+  * Documentation Issues: Since the format for editing graded test follows: `gt/RA1:0 | RA2:0 | MidTerms:0 | Finals:0 | PE:0`
+
+**Alternative 2 :** Using floats for graded test score.
+- Pros:
+  * Numeric Operations: Easier and more efficient numeric operations without the need for extensive parsing.
+- Cons: 
+  * Lesser Flexibility and Customisation: Doesn't allow for special characters
+
+
+**Aspect 2: What should be the inputs for GradedTest Constructor:**
+
+**Alternative 1:** Using Strings for Graded Test Constructor.
+- Pros:
+  * Versatility: Utilizing strings allows for versatile inputs, accommodating various formats and representations.
+  * Customization: Users can input graded test scores in a way that suits their preferences, including default values or special symbols.
+- Cons:
+  * Parsing Overhead: The need for additional parsing introduces complexity, especially when validating inputs.
+  * Type Safety: The reliance on strings may lead to potential type-related issues during runtime.
+  * Complexity: Since all the scores are in a single string, additional parsing will be needed to implement getters and setters.
+  * Learning Curve: Users might need to adapt to a more structured input format, potentially increasing the learning curve.
+  
+**Alternative 2:** Using Structured Object for Graded Test Constructors.
+- Pros:
+  * Type Safety: A structured object ensures type safety, reducing the risk of runtime errors related to data types.
+  * Clarity and Readability: A well-defined object provides clarity and improves the readability of the code, making it easier to understand.
+- Cons:
+  * Less Flexibility: All 5 fields, `ra1, ra2, midterms, finals, pe` needs to be present at all times. Need to create 5 objects for 1 graded test.
+  * Logging/Documentation: Additional parsing is needed to properly store these fields in the json file properly.
+
+**Alternative 3 (current choice):** Use both Strings and Structured Objects for Graded Test Calculators.
+- Pros:
+  * Type Safety Utilization: Ensures type safety with objects and allows the use of strings for flexibility. We can take advantage of the strings to enable the use of `default` values.
+  * Robust Testing: Better bug identification, as the two constructors check each other.
+- Cons:
+  * Extra caution is needed to ensure that both these constructors are compatible with one another. 
+
+
+**Aspect 3: How to store graded test scores for individuals**
```

Comment:
I believe this line should end with a colon, to standardise with Aspect 1 and Aspect 2